docs/faq.md

```diff
@@ -285,4 +285,31 @@ Note: Windows with Radeon GPUs currently default to 1 model maximum due to limit
 
 ## How does Ollama load models on multiple GPUs?
 
-Installing multiple GPUs of the same brand can be a great way to increase your available VRAM to load larger models.  When you load a new model, Ollama evaluates the required VRAM for the model against what is currently available.  If the model will entirely fit on any single GPU, Ollama will load the model on that GPU.  This typically provides the best performance as it reduces the amount of data transfering across the PCI bus during inference.  If the model does not fit entirely on one GPU, then it will be spread across all the available GPUs.
+When loading a new model, Ollama evaluates the required VRAM for the model against what is currently available.  If the model will entirely fit on any single GPU, Ollama will load the model on that GPU.  This typically provides the best performance as it reduces the amount of data transferring across the PCI bus during inference.  If the model does not fit entirely on one GPU, then it will be spread across all the available GPUs.
+
+## How can I enable Flash Attention?
+
+Flash Attention is a feature of most modern models that can significantly reduce memory usage as the context size grows.  To enable Flash Attention, set the `OLLAMA_FLASH_ATTENTION` environment variable to `1` when starting the Ollama server.
+
+> Note: Advanced users using CUDA may benefit from building Ollama and passing `GGML_CUDA_FA_ALL_QUANTS=1` to the llama.cpp build to enable FA for all combinations of quantization types. More information on this can be found in [llama.cpp](https://github.com/ggerganov/llama.cpp/blob/fb4a0ec0833c71cff5a1a367ba375447ce6106eb/ggml/src/ggml-cuda/fattn-common.cuh#L575).
```

Comment:
@sammcj do you know if `GGML_CUDA_FA_ALL_QUANTS` flag is required (or has any impact) to support the 4-bit and 8-bit FA quants we're limiting to below? If it has no impact should we remove this note?

Reply:
The documentation around that build flag's behaviour is ... "not great" 🤣,

I did some digging around on this previously and it seems to be based on the GGML head size.

I came to this conclusion (I'm not sure if it's 100% accurate):

head size = hidden_size/num_attention_heads

| Head Size | Supported K/V Combinations    |
|-----------|-------------------------------|
| 64        | f16/f16 only                  |
| 128       | q4_0/q4_0, q8_0/q8_0, f16/f16 |
| 256       | f16/f16 only                  |

Given most models head size of 128 it should be fine to leave out assuming Ollama is only going to allow Q4_0 or Q8_0 and is not allowing different quants for K vs V.

https://github.com/ggerganov/llama.cpp/blob/master/ggml/src/ggml-cuda/fattn-common.cuh#L575

Reply:
I've removed mention of this from the FAQ for now.